src/WebJobs.Extensions.DurableTask/DurabilityProvider.cs

```diff
@@ -418,5 +433,51 @@ internal virtual bool ConnectionNameMatches(DurabilityProvider durabilityProvide
         {
             return this.ConnectionName.Equals(durabilityProvider.ConnectionName);
         }
+
+        /// <summary>
+        /// Returns the instance id of the entity scheduler for a given entity id.
+        /// </summary>
+        /// <param name="entityId">The entity id.</param>
+        /// <returns>The instance id of the scheduler.</returns>
+        protected string GetSchedulerIdFromEntityId(EntityId entityId)
+        {
+            return EntityId.GetSchedulerIdFromEntityId(entityId);
+        }
+
+        /// <summary>
+        /// Reads the state of an entity from the serialized entity scheduler state.
+        /// </summary>
+        /// <param name="state">The orchestration state of the scheduler.</param>
+        /// <param name="serializerSettings">The serializer settings.</param>
+        /// <param name="result">The serialized state of the entity.</param>
+        /// <returns>true if the entity exists, false otherwise.</returns>
+        protected bool TryGetEntityStateFromSerializedSchedulerState(OrchestrationState state, JsonSerializerSettings serializerSettings, out string result)
+        {
+            if (state != null
+                && state.OrchestrationInstance != null
+                && state.Input != null)
+            {
+                var schedulerState = JsonConvert.DeserializeObject<SchedulerState>(state.Input, serializerSettings);
+
+                if (schedulerState.EntityExists)
+                {
+                    result = schedulerState.EntityState;
+                    return true;
+                }
+            }
+
+            result = null;
+            return false;
+        }
+
+        /// <summary>
+        /// Converts the DTFx representation of the orchestration state into the DF representation.
+        /// </summary>
+        /// <param name="orchestrationState">The orchestration state.</param>
+        /// <returns>The orchestration status.</returns>
+        protected DurableOrchestrationStatus ConvertOrchestrationStateToStatus(OrchestrationState orchestrationState)
+        {
+            return DurableClient.ConvertOrchestrationStateToStatus(orchestrationState);
```

Comment:
Similarly here, I'm not sure I love the idea of adding protected methods to give us access to some internal static method.  Would it be easier to just to make these static methods public (and maybe put them in a more aptly named class if other storage providers need access to this method)?

Reply:
I suppose I did not think of a case where anybody other than the DurabilityProvider implementor needs access.  Though I guess AzureStorage provider is an example where one may want access without going via a DurabilityProvider. Perhaps a static class would be better, such as ProviderUtils so users know not to look at it unless they implement a provider. I will do that.

Reply:
I have pushed this change also.